erpnext/projects/doctype/task/task.py

```diff
@@ -43,6 +44,11 @@ def validate_dates(self):
 		if self.act_start_date and self.act_end_date and getdate(self.act_start_date) > getdate(self.act_end_date):
 			frappe.throw(_("'Actual Start Date' can not be greater than 'Actual End Date'"))
 
+		doc = frappe.get_doc("Project", self.project)
+		if  self.exp_end_date and doc.expected_end_date and getdate(self.exp_end_date) > getdate(doc.expected_end_date) :
+			frappe.msgprint(_("Expected end date cannot be after Project: <b>'{0}'</b> Expected end date").format(doc.name))
```

Comment:
Do a `frappe.throw` instead, with the exception.

Reply:
Done